internal/transform/format.go

```diff
@@ -0,0 +1,492 @@
+type NodeType int
+
+const (
+	RootNode NodeType = iota
+	ComponentsNode
+	PathNode
+	InfoNode
+	OperationNode
+	SchemaNode
+	ResponseNode
+	SecurityNode
+	UrlNode
+	TagNode
+	UnknownNode
+)
+
+func (n NodeType) String() string {
+	return [...]string{
+		"RootNode",
+		"ComponentsNode",
+		"PathNode",
+		"InfoNode",
+		"OperationNode",
+		"SchemaNode",
+		"ResponseNode",
+		"SecurityNode",
+		"UrlNode",
+		"TagNode",
+		"UnknownNode",
+	}[n]
+}
+
+// Define the desired order of keys for different levels
+var rootOrder = []string{
+	"openapi",
+	"info",
+	"externalDocs",
+	"security",
+	"servers",
+	"tags",
+	"paths",
+	"components",
+}
+
+var infoOrder = []string{
+	"title",
+	"version",
+	"summary",
+	"description",
+	"termsOfService",
+	"contact",
+	"license",
+}
+
+var contactOrder = []string{
+	"name",
+	"url",
+	"email",
+}
+
+var componentsOrder = []string{
+	"schemas",
+	"responses",
+	"parameters",
+	"examples",
+	"requestBodies",
+	"headers",
+	"securitySchemes",
+	"links",
+	"callbacks",
+}
+
+var pathOrder = []string{
+	"summary",
+	"description",
+	"servers",
+	"parameters",
+	"get",
+	"post",
+	"put",
+	"delete",
+	"patch",
+	"head",
+	"options",
+	"trace",
+}
+
+var operationOrder = []string{
+	"tags",
+	"summary",
+	"operationId",
+	"description",
+	"externalDocs",
+	"parameters",
+	"requestBody",
+	"responses",
+	"callbacks",
+	"deprecated",
+	"security",
+	"servers",
+}
+
+var schemaOrder = []string{
+	"name",
+	"type",
+	"title",
+	"summary",
+	"description",
+	"in",
+	"$ref",
+	"format",
+	"enum",
+	"default",
+	"multipleOf",
+	"maximum",
+	"exclusiveMaximum",
+	"minimum",
+	"exclusiveMinimum",
+	"maxLength",
+	"minLength",
+	"pattern",
+	"maxItems",
+	"minItems",
+	"uniqueItems",
+	"maxProperties",
+	"minProperties",
+	"required",
+	"properties",
+	"items",
+	"anyOf",
+	"oneOf",
+	"allOf",
+	"not",
+	"additionalProperties",
+	"example",
+	"examples",
+	"deprecated",
+}
+
+var responseOrder = []string{
+	"description",
+	"headers",
+	"content",
+	"links",
+}
+
+var securityOrder = []string{
+	"type",
+	"description",
+	"name",
+	"in",
+	"scheme",
+	"bearerFormat",
+	"flows",
+	"openIdConnectUrl",
+}
+
+var parameterOrder = []string{
+	"name",
+	"in",
+	"description",
+	"required",
+	"deprecated",
+	"allowEmptyValue",
+	"style",
+	"explode",
+	"allowReserved",
+	"schema",
+	"example",
+	"examples",
+	"content",
+}
+
+var requestBodyOrder = []string{
+	"description",
+	"content",
+	"required",
+}
+
+var headerOrder = []string{
+	"description",
+	"required",
+	"deprecated",
+	"allowEmptyValue",
+	"style",
+	"explode",
+	"schema",
+	"example",
+	"examples",
+	"content",
+}
+
+var tagOrder = []string{
+	"name",
+	"description",
+	"externalDocs",
+}
+
+var urlOrder = []string{
+	"name",
+	"identifier",
+	"url",
+	"description",
+	"variables",
+	"email",
+}
+
+var unknownOrder = []string{}
+
+// Orders contains the order of keys for each node type
+type Orders struct {
```

Comment:
Honestly not entirely sure why this is necessary when the individual slices and their existing names are just as clear

Reply:
It was much easier to use this when adding each level, and the logic for determining the order to use completely changed a few times as I realized some silly mistakes in the initial implementation. 

Happy to simplify if needed.